### ISSUE TITLE
Error TypeError: pie() got an unexpected keyword argument 'normalize'

### ISSUE THREAD

**Issue by csoscd** [molu8bits/modsecurity-parser#14]
Hi,

I just updated to the most recent version on my server (running Ubuntu 18.04 LTS). Unfortunatelly I get the following error:

Traceback (most recent call last):
  File "/opt/modsecurity-parser-master/modsecurity-parser.py", line 689, in <module>
    outputWithGraphs = modsecViewGraphs(modsec_entries)
  File "/opt/modsecurity-parser-master/modsecurity-parser.py", line 454, in modsecViewGraphs
    patches, texts, autotexts = plt.pie(intercepted_cnt_top10.values(), autopct='%1.1f%%', shadow=True, startangle=90, radius=1.0, normalize=True)
TypeError: pie() got an unexpected keyword argument 'normalize'

Any ideas?

**Comment by molu8bits** [comment]
Hi,
What is the  version of matplotlib ? can you try to upgrade it with pip ?
How to check version:
https://matplotlib.org/stable/faq/troubleshooting_faq.html

**Comment by miller45** [comment]
I get the same error and pip3 installed version 3.1.2 of matplotlib today

**Comment by miller45** [comment]
after doing `pip3 install --upgrade --force-reinstall matplotlib` i have now version 3.4.3 of matplotlib.
That yields now another error:
```
modsecurity-parser.py:437: MatplotlibDeprecationWarning: Passing non-integers as three-element position specification is deprecated since 3.3 and will be removed two minor releases later.
  plt.subplot(ax21)
Traceback (most recent call last):
  File "modsecurity-parser.py", line 689, in <module>
    outputWithGraphs = modsecViewGraphs(modsec_entries)
  File "modsecurity-parser.py", line 437, in modsecViewGraphs
    plt.subplot(ax21)
  File "/home/rklein/.local/lib/python3.8/site-packages/matplotlib/pyplot.py", line 1272, in subplot
    key = SubplotSpec._from_subplot_args(fig, args)
  File "/home/rklein/.local/lib/python3.8/site-packages/matplotlib/gridspec.py", line 632, in _from_subplot_args
    raise ValueError(
ValueError: Single argument to subplot must be a three-digit integer, not AxesSubplot(0.125,0.420962;0.149038x0.148077)

```

**Comment by miller45** [comment]
Ok but this can be resolved see: #16
It basically means you need exactly verson 3.3.2...this should be added to requirements.txt
